lib/services/front.ts

```diff
@@ -1,357 +1,127 @@
-				return conversation.subject === subject;
-			});
-			// Return the most recent, if any
-			if (conversationsMatched.length > 0) {
-				return conversationsMatched[0].id;
-			}
-			// Recurse up to the specified number of times
-			if (attemptsLeft > 1) {
-				return this.findConversation(subject, attemptsLeft - 1);
-			}
-			throw new Error('Could not find relevant conversation.');
-		});
-	}
-
-	/**
-	 * The name of this service, as required by the framework.
-	 * @returns  'flowdock' string.
-	 */
-	get serviceName(): string {
-		return FrontService._serviceName;
-	}
-
-	/**
-	 * Retrieve the SDK API handle for Front.
-	 * @return  The Front SDK API handle.
-	 */
-	get apiHandle(): FrontHandle {
-		return {
-			front: this.session
-		};
-	}
-}
-
-/**
- * Build this class, typed and activated as a listener.
- * @returns  Service Listener object, awakened and ready to go.
- */
-export function createServiceListener(data: FrontConstructor): ServiceListener {
-	return new FrontService(data, true);
-}
-
-/**
- * Build this class, typed as an emitter.
- * @returns  Service Emitter object, ready for your events.
- */
-export function createServiceEmitter(data: FrontConstructor): ServiceEmitter {
-	return new FrontService(data, false);
-}
-
-/**
- * Build this class, typed as a message service.
- * @returns  Message Service object, ready to convert events.
- */
-export function createMessageService(data: FrontConstructor): Messenger {
-	return new FrontService(data, false);
-}
+// /*
+//  Copyright 2016-2017 Resin.io
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  */
+//
+// import * as Promise from 'bluebird';
+// import { Front } from 'front-sdk';
+// import * as path from 'path';
+//
+// import {
+// 	FrontConnectionDetails, FrontEmitContext,
+// 	FrontEvent, FrontHandle, FrontResponse,
+// } from './front-types';
+// import { ServiceEmitter, ServiceListener } from './service-types';
+// import { ServiceUtilities } from './service-utilities';
+//
+// export class FrontService extends ServiceUtilities<string> implements ServiceListener, ServiceEmitter {
+// 	private static _serviceName = path.basename(__filename.split('.')[0]);
+//
+// 	/** Underlying SDK object that we route requests to */
+// 	private session: Front;
+//
+// 	constructor(data: FrontConnectionDetails, listen: boolean) {
+// 		super();
+// 		this.session = new Front(data.token);
+// 		if (listen) {
```

Comment:
I don't think there's much need for a separate `startListening()` method. The constructor for most services is going to be small enough that adding the code under `if (listen) {` is fine.